pallets/entry/src/lib.rs

```diff
@@ -615,4 +626,21 @@ impl<T: Config> Pallet<T> {
 			.map_err(|_| Error::<T>::StateUpdateFailed)?;
 		Ok(())
 	}
+
+	pub fn verify_digest(
```

Comment:
Optimise this. for eg:
```
pub fn verify_digest(
    digest: T::Hash,
    registry_id: Option<RegistryIdOf>,
) -> Result<Option<RegistryEntryIdOf>, Error<T>> {
    let result = if let Some(reg_id) = registry_id {
        // Direct lookup: O(1)
        HashToIdentifier::<T>::get(&digest, reg_id)
    } else {
        // Prefix lookup: stop at first match (O(1) in practice, instead of O(n))
        HashToIdentifier::<T>::iter_prefix(&digest)
            .next()
            .map(|(_reg_id, entry_id)| entry_id)
    };

    Ok(result)
}
```

Reply:
Updated